backend/dashboard/calculations.py

```diff
@@ -3,211 +3,212 @@
+            return ESG_scores[company_name]['environment_score']
+        else:
+            return 0
+
+    def _is_green(transaction: dict, ESG_scores: dict[str,dict]):
         company_env_score = _get_company_env_score(transaction, ESG_scores)
-        top_5_companies.append({
-            'Company Name': transaction['merchant_name'],
-            'ESG Score': company_env_score,
-            'Amount Spent': transaction['amount']
-        })
-    
-    return top_5_companies
-
-
-def calculate_historical_scores(transactions: dict[int, dict], esg_scores: dict[str, dict]) -> list[int]:
-    """
-    Return list of environmental scores for the past 12 months.
-    The scores go from most to least recent! So scores[0] is this month, scores[-1] is 10 months ago
-    """
-    scores = []
-    current_date =  datetime.now() - relativedelta(months=1)
-    delta_days = 30
-
-    for _ in range(12):
-        # Get start and end dates
-        start_date, end_date = _get_start_end_dates(current_date)
+        return company_env_score > 500
+
+    def _company_tier(company_env_score: int) -> int:
+        """
+        Returns the tier of the company based on its environmental score, 
+        worst tier is 4 and best tier is 1.
+        """
+        if company_env_score > 560:
+            return 1
+        elif company_env_score > 520:
+            return 2
+        elif company_env_score > 500:
+            return 3
+        else:
+            return 4
         
-        # Calculate the score for the current period
-        score = calculate_score(transactions, start_date, end_date, esg_scores)
+
+    def _get_start_end_dates(current_date: datetime) -> tuple[datetime]:
+        """
+        Helper function, get start and end dates of the month of the current_date.
+        """
+        # Get the first day of the month
+        start_date = current_date.replace(day=1)
+        # Get the last day of the month
+        last_day = monthrange(current_date.year, current_date.month)[1]
+        end_date = current_date.replace(day=last_day)
         
-        scores.append(score)
+        return start_date, end_date
+
+
+
+    def _count_green_transactions_in_period(transactions: list[dict], start_date: datetime, end_date: datetime, ESG_scores: dict[str,dict]) -> int:
+        """
+        Helper function that counts green transactions within a specified date range.
+        """
+        return sum(
+            1 for transaction in transactions
+            if start_date <= datetime.strptime(transaction['time_completed'], "%Y-%m-%dT%H:%M:%S.%fZ") <= end_date
+            and _is_green(transaction, ESG_scores)
+        )
+
+
+    def _get_unique_companies(transactions: list[dict], ESG_scores: dict[str, dict]):
+        """
+        Return set of top companies shopped at for all time
+        """
+        unique_companies = set()
+
+        for transaction in transactions:
+            company_name = _get_closest_match(transaction['merchant_name'], ESG_scores)
```

Comment:
These helper functions aren't recognized now that they're within a class. I think you need to add the self parameter into each other methods, and then call them using self._get_closest_match